builtin/providers/aws/resource_aws_vpn_gateway.go

```diff
@@ -80,7 +81,7 @@ func resourceAwsVpnGatewayRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	vpnGateway := resp.VpnGateways[0]
-	if vpnGateway == nil {
+	if vpnGateway == nil || *vpnGateway.State == "deleted" {
```

Comment:
if we are adding state == "deleted" here, then we should remove it from line 89 :)